packages/react-components/react-tree/src/components/TreeItem/useTreeItem.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { resolveShorthand } from '@fluentui/react-utilities';
+import type { TreeItemElement, TreeItemProps, TreeItemState } from './TreeItem.types';
+import { ChevronRightRegular } from '@fluentui/react-icons';
+import { useFluent_unstable } from '@fluentui/react-shared-contexts';
+import { useBaseTreeItem_unstable } from '../BaseTreeItem/index';
+
+/**
+ * Create the state required to render TreeItem.
+ *
+ * The returned state can be modified with hooks such as useTreeItemStyles_unstable,
+ * before being passed to renderTreeItem_unstable.
+ *
+ * @param props - props from this instance of TreeItem
+ * @param ref - reference to root HTMLElement of TreeItem
+ */
+export const useTreeItem_unstable = (props: TreeItemProps, ref: React.Ref<TreeItemElement>): TreeItemState => {
+  const treeItemState = useBaseTreeItem_unstable(props, ref);
+  const { expandIcon, beforeIcon, afterIcon, actionIcon } = props;
+  const { dir } = useFluent_unstable();
+  const expandIconRotation = treeItemState.open ? 90 : dir !== 'rtl' ? 0 : 180;
+  return {
+    ...treeItemState,
+    components: {
+      ...treeItemState.components,
+      expandIcon: 'span',
+      beforeIcon: 'span',
+      afterIcon: 'span',
+      actionIcon: 'span',
+    },
+    beforeIcon: resolveShorthand(beforeIcon, {
+      defaultProps: { 'aria-hidden': true },
+    }),
+    afterIcon: resolveShorthand(afterIcon, {
+      defaultProps: { 'aria-hidden': true },
+    }),
+    expandIcon: resolveShorthand(expandIcon, {
+      required: !treeItemState.isLeaf,
+      defaultProps: {
+        children: <ChevronRightRegular style={{ transform: `rotate(${expandIconRotation}deg)` }} />,
```

Comment:
is the rotation necessary? there is no other chevron icon in v9?

Reply:
remember that the icon should handle rtl

Reply:
I'm just following `AccordionHeader` implementation. The rotation mechanism uses `rtl` from `useFluent` to properly rotate, I think rotating might be a smaller impact in the bundlesize and also more appropriate for future animations?!

Reply:
that actually makes sense 👍